kolibri/plugins/coach/assets/src/composables/useResourceSelection.js

```diff
@@ -0,0 +1,155 @@
+ * and topic trees, and offers methods to add, remove, or override selected resources.
+ *
+ * @typedef {Object} UseResourceSelectionResponse
+ * @property {Object} topic Topic tree object, contains the information of the topic,
+ *   its ascendants and children.
+ *   Defined only if the `topicId` query in the route is set.
+ * @property {boolean} loading Indicates whether the main topic tree, channels, and bookmarks
+ *   data are currently loading. This does not account for loading more data. For such cases,
+ *   use the fetch objects of each entity.
+ * @property {FetchObject} channelsFetch Channels fetch object to manage the process of
+ *   fetching channels. We currently don't support fetching more channels.
+ * @property {FetchObject} bookmarksFetch Bookmarks fetch object to manage the process of
+ *   fetching bookmarks. Fetching more bookmarks is supported.
+ * @property {FetchObject} treeFetch Topic tree fetch object to manage the process of
+ *   fetching topic trees and their resources. Fetching more resources is supported.
+ * @property {Array<(node: Object) => boolean>} selectionRules An array of functions that determine
+ *   whether a node can be selected.
+ * @property {Array<Object>} selectedResources An array of currently selected resources.
+ * @property {(resources: Array<Object>) => void} selectResources Adds the specified resources
+ *   to the `selectedResources` array.
+ * @property {(resources: Array<Object>) => void} deselectResources Removes the specified resources
+ *   from the `selectedResources` array.
+ * @property {(resources: Array<Object>) => void} setSelectedResources Replaces the current
+ *   `selectedResources` array with the provided resources array.
+ *
+ * @returns {UseResourceSelectionResponse}
+ */
+export default function useResourceSelection() {
+  const store = getCurrentInstance().proxy.$store;
+  const route = computed(() => store.state.route);
+  const topicId = computed(() => route.value.query.topicId);
+
+  const selectionRules = ref([]);
+  const selectedResources = ref([]);
+
+  const bookmarksFetch = useFetch({
+    fetchMethod: () =>
+      ContentNodeResource.fetchBookmarks({
+        params: { limit: 25, available: true },
+      }),
+    fetchMoreMethod: more =>
+      ContentNodeResource.fetchBookmarks({
+        params: more,
+      }),
+    dataKey: 'results',
+    moreKey: 'more',
+    additionalDataKeys: {
+      count: 'count',
+    },
+  });
+
+  const channelsFetch = useFetch({
+    fetchMethod: () =>
+      ChannelResource.fetchCollection({
+        getParams: {
+          available: true,
+        },
+      }),
+  });
+
+  const treeFetch = useFetch({
+    fetchMethod: () =>
+      ContentNodeResource.fetchTree({ id: topicId.value, params: { include_coach_content: true } }),
+    fetchMoreMethod: more => ContentNodeResource.fetchTree({ id: topicId.value, params: more }),
+    dataKey: 'children.results',
+    moreKey: 'children.more.params',
```

Comment:
`children.more` just needs to be passed directly to the `fetchTree` method of `ContentNodeResource`.

Reply:
oh, yes :sweat_smile:. I was imitating the `useFetchTree` and didnt notice this

Reply:
Updated!